src/js/crud.js

```diff
@@ -0,0 +1,136 @@
+        e.preventDefault();
+    });
+  }
+  // Sets an event listener for adding task when enter button is pressed
+  AddTaskPressEnter() {
+    this.inputField.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        this.addTask(this.inputField.value);
+        this.inputField.value = '';
+      } 
+    });
+  }
+  //Sorts the tasks based on their index
+  sortTasks() {
+    this.todoTask.sort ((a, b) => a.index - b.index)
+  }
+  // Refreshes the tasks presented in the UI
+  render() {
+    this.sortTasks();
+    this.todoList.innerHTML = '';
+    
+    this.todoTask.forEach((task, i) => {
+      let taskElement = document.createElement('li');
+      taskElement.classList.add('task-container');
+  
+      let taskContent = document.createElement('div');
+      taskContent.classList.add('task');
+      
+      let checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.checked = task.completed;
+      
+      taskContent.appendChild(checkbox);
+    
+      let description = document.createElement('span');
+      description.classList.add('description');
+      description.contentEditable = 'true';
+      description.textContent = task.description;
+      description.addEventListener('input', () => {
+        this.editTask(i, description.textContent); 
+      });
+      taskContent.appendChild(description);
+    
+      let deleteIcon = document.createElement('span');
+      deleteIcon.innerHTML = 'delete';
+      deleteIcon.classList.add('material-icons', 'trash-icon');
+      deleteIcon.style.display = 'none';
+      deleteIcon.dataset.task = i;
+      
+      let moreVertIcon = document.createElement('span');
+      moreVertIcon.innerHTML = 'more_vert';
+      moreVertIcon.classList.add('material-icons', 'dots-icon');
+      
+      taskElement.append(taskContent, deleteIcon, moreVertIcon);
+      
+      // Add the task element to the to-do list.
+      this.todoList.appendChild(taskElement);
+      
+      taskElement.addEventListener('focus', (event) => {
+        event.currentTarget.querySelector('.dots-icon').style.display = 'none';
+        event.currentTarget.querySelector('.trash-icon').style.display = '';
+        event.currentTarget.style.backgroundColor = 'lightyellow';
+      }, true);
+      
+      taskElement.addEventListener('blur', (event) => {
+        event.currentTarget.querySelector('.dots-icon').style.display = '';
+        event.currentTarget.querySelector('.trash-icon').style.display = 'none';
+        event.currentTarget.style.backgroundColor = '';
+      }, true);
+    });
+  
+    // Persist the updates to local storage.
+    this.updateLocalStorage();
+  }
+  //Adds a new task with the supplied description, updates local storage and the UI.
+  addTask(description) {
+    if (description.trim() !== '') {
+      const newTask = { 
+        description,
+        completed: false,
+        index: this.todoTask.length,
+      };
```

Comment:
- [ ] Please makes sure that property `index` set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). However as you can see in the image below, your first task has `index` to be `0` which is suppose to be 1 and the second task too to have `index` to be `1` which is suppose to be `2` :blush: 
  
  ![Screenshot from 2023-07-14 23-20-32](https://github.com/Rebelzob/To-do-app/assets/73854999/08c0bc4f-93da-43e7-ad61-b723f7493a24)